integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py

```diff
@@ -292,10 +306,22 @@ def run(self, messages: List[ChatMessage]):
             content=new_message,
             generation_config=self._generation_config,
             safety_settings=self._safety_settings,
+            stream=streaming_callback is not None,
         )
 
+        replies = self.get_stream_response(res, streaming_callback) if streaming_callback else self.get_response(res)
```

Comment:
```suggestion
        replies = self._get_stream_response(res, streaming_callback) if streaming_callback else self._get_response(res)
```

Let's make these methods internal